src/unit-stack-configuration/unit-stack-configuration.repository.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { SelectQueryBuilder } from 'typeorm';
+import { UnitStackConfigurationRepository } from './unit-stack-configuration.repository';
+
+const unitStackConfiguration = new UnitStackConfiguration();
+
+const mockQueryBuilder = () => ({
+  innerJoinAndSelect: jest.fn(),
+  innerJoin: jest.fn(),
+  where: jest.fn(),
+  andWhere: jest.fn(),
+  getMany: jest.fn(),
+  getOne: jest.fn(),
+});
+
+describe('UnitStackConfigurationRepository', () => {
+  let repository;
+  let queryBuilder;
+
+  beforeEach(async () => {
+    const module = await Test.createTestingModule({
+      providers: [
+        UnitStackConfigurationRepository,
+        { provide: SelectQueryBuilder, useFactory: mockQueryBuilder },
+      ],
+    }).compile();
+
+    repository = module.get(UnitStackConfigurationRepository);
+    queryBuilder = module.get<
+      SelectQueryBuilder<UnitStackConfigurationRepository>
+    >(SelectQueryBuilder);
+  });
+
+  describe('getUnitStackConfigsByLocationIds', () => {
+    it('calls createQueryBuilder and gets all Unit Stack Configurations from the repository', async () => {
+      repository.createQueryBuilder = jest.fn().mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.innerJoin.mockReturnValue(queryBuilder);
+      queryBuilder.innerJoin.mockReturnValue(queryBuilder);
+      queryBuilder.where.mockReturnValue(queryBuilder);
+      queryBuilder.andWhere.mockReturnValue(queryBuilder);
+      queryBuilder.getMany.mockReturnValue([unitStackConfiguration]);
+
+      const result = await repository.getUnitStackConfigsByLocationIds(
+        ["1"]
+      );
+
+      expect(result).toEqual([unitStackConfiguration]);
+    });
+  });
+
+  describe('getUnitStackConfigsByUnitId', () => {
+    it('calls createQueryBuilder and gets all Unit Stack Configations from the repository when it is a unit', async () => {
+      repository.createQueryBuilder = jest.fn().mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.where.mockReturnValue(queryBuilder);
+
+      queryBuilder.getMany.mockReturnValue([unitStackConfiguration]);
+
+      const result = await repository.getUnitStackConfigsByUnitId('1', true);
+
+      expect(result).toEqual([unitStackConfiguration]);
+    });
+
+    it('calls createQueryBuilder and gets all Unit Stack Configations from the repository when it is not a unit', async () => {
+      repository.createQueryBuilder = jest.fn().mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.innerJoinAndSelect.mockReturnValue(queryBuilder);
+      queryBuilder.where.mockReturnValue(queryBuilder);
+
+      queryBuilder.getMany.mockReturnValue([unitStackConfiguration]);
```

Comment:
You should only need to mock getMany, getOne, etc. in each test since that is the only method that varies in it's return value between tests.  Each of the other queryBuilder methods that just returns the queryBuilder should be able to be mocked once globally.

Reply:
it was giving some weird errors when mocked globally but I did reduce the amount of mocks